lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResult.cs

```diff
@@ -98,15 +105,21 @@ public string GetUrlResult(string url)
 
         public void WaitForDriverToFinish()
         {
-            DriverStatus status = FetchDriverStatus();
+            DriverStatus status = FetchFirstDriverStatus();
+
+            if (DriverStatus.UNKNOWN == status)
+            {
+                // We were unable to connect to the Driver at least once.
+                throw new WebException("Unable to connect to the Driver.");
+            }
             
             while (status.IsActive())
             {
                 try
                 {
                     status = FetchDriverStatus();
                 }
-                catch (System.Net.WebException)
+                catch (WebException)
                 {
                     // If we no longer can reach the Driver, it must have exited.
                     status = DriverStatus.UNKNOWN_EXITED;
```

Comment:
Shall we write something to the log?

Reply:
Hmm, it is entirely OK for the Driver to go down without notice within this API. Hence, I'd rather not.